Gifski/ConversionCompletedView.swift

```diff
@@ -128,3 +141,44 @@ final class ConversionCompletedView: SSView {
 		])
 	}
 }
+
+extension ConversionCompletedView: QLPreviewPanelDataSource {
+	override func acceptsPreviewPanelControl(_ panel: QLPreviewPanel!) -> Bool {
+		return true
+	}
+
+	override func beginPreviewPanelControl(_ panel: QLPreviewPanel!) {
+		panel.delegate = self
+		panel.dataSource = self
+	}
+
+	override func endPreviewPanelControl(_ panel: QLPreviewPanel!) {
+		panel.dataSource = nil
+		panel.delegate = nil
+	}
+
+	func numberOfPreviewItems(in panel: QLPreviewPanel!) -> Int {
+		return 1
+	}
+
+	func previewPanel(_ panel: QLPreviewPanel!, previewItemAt index: Int) -> QLPreviewItem! {
+		return fileUrl as NSURL
+	}
+}
+
+extension ConversionCompletedView: QLPreviewPanelDelegate {
+	func previewPanel(_ panel: QLPreviewPanel!, sourceFrameOnScreenFor item: QLPreviewItem!) -> NSRect {
+		guard let imageRect = draggableFile.subviews.first?.frame, let window = window else {
+			return .zero
+		}
+
+		let windowFrame = draggableFile.convert(imageRect, to: nil)
+		let screenFrame = window.convertToScreen(windowFrame)
+		
+		return screenFrame
+	}
```

Comment:
I would use this extension:

```swift
extension NSView {
	/// Get the view frame in screen coordinates.
	var frameInScreenCoordinates: CGRect? {
		return window?.convertToScreen(convert(bounds, to: nil))
	}
}
```

Then the above can simply be:

```swift
func previewPanel(_ panel: QLPreviewPanel!, sourceFrameOnScreenFor item: QLPreviewItem!) -> NSRect {
	return draggableFile.subviews.first?.frameInScreenCoordinates ?? .zero
}
```

Reply:
Good point as well 👍